s4e-backend/src/main/java/pl/cyfronet/s4e/controller/GroupController.java

```diff
@@ -0,0 +1,177 @@
+import io.swagger.annotations.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import pl.cyfronet.s4e.bean.Group;
+import pl.cyfronet.s4e.controller.response.GroupResponse;
+import pl.cyfronet.s4e.event.OnAddToGroupEvent;
+import pl.cyfronet.s4e.event.OnRemoveFromGroupEvent;
+import pl.cyfronet.s4e.ex.BadRequestException;
+import pl.cyfronet.s4e.ex.GroupCreationException;
+import pl.cyfronet.s4e.ex.GroupUpdateException;
+import pl.cyfronet.s4e.ex.NotFoundException;
+import pl.cyfronet.s4e.service.AppUserService;
+import pl.cyfronet.s4e.service.GroupService;
+import pl.cyfronet.s4e.service.InstitutionService;
+import pl.cyfronet.s4e.service.SlugService;
+
+import javax.validation.Valid;
+import java.util.stream.Collectors;
+
+import static pl.cyfronet.s4e.Constants.API_PREFIX_V1;
+
+@RestController
+@RequestMapping(API_PREFIX_V1)
+@RequiredArgsConstructor
+public class GroupController {
+    private final GroupService groupService;
+    private final AppUserService appUserService;
+    private final InstitutionService institutionService;
+    private final ApplicationEventPublisher eventPublisher;
+    private final SlugService slugService;
+
+    @ApiOperation("Create a new group")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "If group was created"),
+            @ApiResponse(code = 400, message = "Group not created"),
+            @ApiResponse(code = 404, message = "Institution not found")
+    })
+    @PostMapping("/institutions/{institution}/groups")
+    public ResponseEntity<?> create(@RequestBody @Valid String name,
+                                    @PathVariable("institution") String institutionSlug)
+            throws GroupCreationException, NotFoundException {
+        val institution = institutionService.getInstitution(institutionSlug)
+                .orElseThrow(() -> new NotFoundException("Institution not found for id '" + institutionSlug));
+        Group group = Group.builder().name(name).slug(slugService.slugify(name)).institution(institution).build();
+        groupService.save(group);
+        return ResponseEntity.ok().build();
+    }
+
+    @ApiOperation("Add a new member to the group")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "If member was added"),
+            @ApiResponse(code = 400, message = "Member not added"),
+            @ApiResponse(code = 404, message = "Group or user not found")
+    })
+    @PostMapping("/institutions/{institution}/groups/{group}/members")
+    public ResponseEntity<?> addMember(@PathVariable("group") String groupSlug,
+                                       @RequestBody Long userId,
```

Comment:
Same as with the previous method, a request object would be nicer I guess, also parameters are ordered in a weird way